aea/helpers/dialogue/base.py

```diff
@@ -830,106 +849,88 @@ def update(self, message: Message) -> Optional[Dialogue]:
         If the message is addressed to an existing dialogue, the dialogue is retrieved, extended with this message and returned.
         If there are any errors, e.g. the message dialogue reference does not exists or the message is invalid w.r.t. the dialogue, return None.
 
-        :param message: a new message
+        :param message: a new incoming message
         :return: the new or existing dialogue the message is intended for, or None in case of any errors.
         """
-        dialogue_reference = message.dialogue_reference
+        assert self.is_message_by_other(
+            message
+        ), "Invalid 'update' usage. Update must only be used with a message by another agent."
+        assert message.has_to, "The message's 'to' field is not set {}".format(message)
+        # the message's sender field is ensured to have been set via the is_message_by_other call above
```

Comment:
You can remove this check; it will automatically raise if the to field is accessed and not present; this check adds nothing here

Reply:
Done

Reply:
I see; in this case I was wrong and we should keep it indeed.